lib/virtio/virtqueue.c

```diff
@@ -637,6 +760,11 @@ static int vq_ring_must_notify(struct virtqueue *vq)
 		if (vq->vq_dev->role == VIRTIO_DEV_MASTER) {
 			new_idx = vq->vq_ring.avail->idx;
 			prev_idx = new_idx - vq->vq_queued_cnt;
+
+#ifdef VIRTQUEUE_CACHE_OPS
```

Comment:
do you need to re invalidate ? probably already done when get buffer for TX ( especially if you invalidate the whole structure)